jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/PathResource.java

```diff
@@ -247,6 +258,96 @@ public int hashCode()
         return Objects.hashCode(path);
     }
 
+    @Override
+    public Resource resolve(String subUriPath)
+    {
+        // Check that the path is within the root,
+        // but use the original path to create the
+        // resource, to preserve aliasing.
+        // TODO do a URI safe encoding?
+        if (URIUtil.isNotNormalWithinSelf(subUriPath))
+            throw new IllegalArgumentException(subUriPath);
+
+        if (URIUtil.SLASH.equals(subUriPath))
+            return this;
+
+        // Sub-paths are always resolved under the given URI,
+        // we compensate for input sub-paths like "/subdir"
+        // where default resolve behavior would be to treat
+        // that like an absolute path.
+        while (subUriPath.startsWith(URIUtil.SLASH))
+        {
+            // TODO XXX this appears entirely unnecessary and inefficient.  We already have utilities
+            //      to handle appending path strings with/without slashes.
+            subUriPath = subUriPath.substring(1);
+        }
+
+        URI uri = getURI();
+        URI resolvedUri = URIUtil.addPath(uri, subUriPath);
+        Path path = Paths.get(resolvedUri.normalize());
```

Comment:
why is the normalize here?   won't that hide some aliases?

Reply:
This is because of differences when using `Paths.get(URI)` (super strict) vs `Paths.get(String)` (very flexible).
I'll be happy to show you.

Reply:
This won't hide aliases, as the user provided URI is still tracked separately from the Path in the PathResource.
The URI comparison later, during the alias check will still show the difference we are looking for.

Reply:
We should review all normalize within the Resource layer.

See the other PR about this -> https://github.com/eclipse/jetty.project/pull/8631#discussion_r994087669